### ISSUE TITLE
Custom component not showing error until save is clicked

### ISSUE THREAD

**Issue by g8anusha** [adfinis-sygroup/ember-validated-form#49]
Hi @czosel & others!

I have a form in which I have few custom components (power-select, power-select-multiple, etc.) and validated-form fields.

When I focus on a validated-form's field, and focusout, without value, it shows as a validation error.

But, when I focus on a custom component & focusout without value, it doesn't.

However, if I click on Save, model.validate() is invoked, and thus, the validation error(s) are thrown.

My expectation : Even with custom components, I want to see validation error when I focus out.

Please see this twiddle, for the issue reproduction.
https://ember-twiddle.com/7693dda4ebdebcda2065c0b3087011c5?openFiles=templates.application.hbs%2C

Is it possible to get this done? Please help me with this.

**Comment by g8anusha** [comment]
Sorry, some issue with the twiddle. Twiddle not saving my code. Please find the template code.

```
{{#validated-form
	model=(changeset model UserValidations)
  on-submit    = (action "submit")
  submit-label = 'Save' as |f|}}

  {{f.input label="First name" name="firstName"}}
  {{#f.input label="Last name" name="lastName" as |fi|}}
    {{textarea name="lastName" onupdate=fi.update onhover=fi.setDirty
  {{/f.input}}
  {{f.submit label="Save" disabled=f.model.isInvalid}}

{{/validated-form}}
```

**Comment by czosel** [comment]
Hi @g8anusha,

the key to achieving this behaviour is calling `fi.setDirty` when your component goes out of focus. Getting this right can sometimes be tricky because how the event is triggered depends on the component you're using: `onhover` is for example just used in our demo app and won't work with textareas. When you're using the ember `{{textarea}}` you'll have to use `input` and `focusout` (see [ember docs](https://www.emberjs.com/api/ember/2.14.1/classes/Ember.Templates.helpers/methods/textarea?anchor=textarea&show=inherited,protected,private,deprecated)), with a native `<textarea>` it would be `oninput` and `onblur`. 

I created a [twiddle](https://ember-twiddle.com/7693dda4ebdebcda2065c0b3087011c5?openFiles=templates.application.hbs%2C) with a working demo.

So, to summarize: It really depends on what kind of custom component you're using. Textareas might not be a perfect example since we support them out of the box anyway. For debugging purposes, I'd recomment to bind you event listener to a custom action to see if it actually gets fired before changing it to call `fi.update` or `fi.setDirty`.

Hope that helps!

PS: My current workflow with ember twiddle is to name the twiddle, then hit File > save to GitHub Gist, and finally open the current page in a new Tab to see if everything actually has been saved.

**Comment by g8anusha** [comment]
Worked like magic! Thanks a lot! :100:
